Source/WebKit/GPUProcess/media/cocoa/RemoteMediaPlayerProxyCocoa.mm

```diff
@@ -99,20 +103,33 @@
     m_layerHostingContextIDRequests.append(WTFMove(completionHandler));
 }
 
-void RemoteMediaPlayerProxy::setVideoLayerSizeFenced(const WebCore::FloatSize& size, WTF::MachSendRight&& machSendRight)
+void RemoteMediaPlayerProxy::setVideoLayerSizeFenced(const WebCore::FloatSize& size, WTF::MachSendRight&& machSendRight, CompletionHandler<void()>&& completionHandler)
 {
     ALWAYS_LOG(LOGIDENTIFIER, size.width(), "x", size.height());
-    if (m_inlineLayerHostingContext)
+
+#if USE(EXTENSIONKIT)
+    RetainPtr<_SEHostingUpdateCoordinator> hostingUpdateCoordinator;
+#endif
+
+    if (m_inlineLayerHostingContext) {
+#if USE(EXTENSIONKIT)
+        auto xpcRepresentation = adoptOSObject(xpc_dictionary_create(nullptr, nullptr, 0));
+        xpc_dictionary_set_mach_send(xpcRepresentation.get(), machPortKey, machSendRight.sendRight());
+        hostingUpdateCoordinator = adoptNS([alloc_SEHostingUpdateCoordinatorInstance() initFromXPCRepresentation:xpcRepresentation.get()]);
+        [hostingUpdateCoordinator addHostable:m_inlineLayerHostingContext->hostable().get()];
+#else
         m_inlineLayerHostingContext->setFencePort(machSendRight.sendRight());
+#endif
+    }
 
     m_configuration.videoLayerSize = size;
     setVideoLayerSizeIfPossible(size);
 
-    m_player->setVideoLayerSizeFenced(size, WTFMove(machSendRight));
+    m_player->setVideoLayerSizeFenced(size, WTFMove(machSendRight), [] { });
```

Comment:
Do we want to pass a no-op completion handler here? Does our completion handler not need to run after the player sets its layer size?

Reply:
I wonder if the below code for committing the hosting update coordinator should be the completion handler for `m_player->setVideoLayerSizeFenced`.

Reply:
You are completely right, this has been addressed in the latest version of the patch.